Framework/LiveData/src/Kafka/IKafkaStreamDecoder.cpp

```diff
@@ -495,6 +502,15 @@ void IKafkaStreamDecoder::checkRunEnd(
   }
 }
 
+int IKafkaStreamDecoder::runNumber() const noexcept {
+  if (m_runId.empty() ||
+      (std::find_if_not(m_runId.cbegin(), m_runId.cend(), [](const char c) {
+         return std::isdigit(c);
+       }) == m_runId.end()))
+    return -1;
+
+  return std::atoi(m_runId.c_str());
```

Comment:
Why is `m_runId` stored as a string, given that we expect/guarantee to convert it to an `int` given `noexcept`?

Reply:
This is consistent with the new scheme where a `runId` may be something other than a string. This method returns -1 if `m_runId` does not contain digits.

Reply:
Yes, sorry missed the return above.

Reply:
Is the logic here correct? I'm reading this and it looks like it returns `-1` only when the run number is convertible to an integer.

e.g.:
```
Breakpoint 4, Mantid::LiveData::IKafkaStreamDecoder::runNumber (this=0x7fff80007220)
    at /home/dan/mantid/Framework/LiveData/src/Kafka/IKafkaStreamDecoder.cpp:534
534	    return -1;
(gdb) p m_runId 
$3 = "233"
(gdb) p m_runId.size()
$4 = 3
```